Makefile

```diff
@@ -1,15 +1,18 @@
-TARGETS=$(patsubst dates/%.yaml, dates/%/ALL, $(wildcard dates/*.yaml))
-ALL: ${TARGETS}
+GENERATOR='./generate.rb'
+
+MAKE_TARGETS=$(patsubst dates/%.yaml, dates/%/ALL, $(wildcard dates/*.yaml))
+CLEAN_TARGETS=$(patsubst dates/%.yaml, dates/%/*, $(wildcard dates/*.yaml))
+ALL: ${MAKE_TARGETS}
 
 clean:
-	rm -rf ${TARGETS}
+	rm -rf ${CLEAN_TARGETS}
 
 # Keep Makefiles
-.SECONDARY: $(patsubst dates/%/ALL, dates/%/Makefile, $(TARGETS))
+.SECONDARY: $(patsubst dates/%/ALL, dates/%/Makefile, $(MAKE_TARGETS))
 
 dates/%/Makefile: dates/%.yaml $(wildcard decks/*.yaml)
 	mkdir -p $(shell dirname $@)
-	./generate.rb $< $(shell dirname $@)
+	$(GENERATOR) $< $(shell dirname $@) || printf "\033[1;31minterpreter or script missing\033[0m\n"
```

Comment:
it was itended to make the user aware of the reason as it might be used of people not familiar with Make, LaTeX nor Ruby or alike.